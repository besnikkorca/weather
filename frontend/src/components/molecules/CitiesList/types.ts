import { City } from 'templates/WeatherAppTemplate/types';

export interface Props {
  active: string;
  setActive: (active: string) => void;
  cities: City[];
}
