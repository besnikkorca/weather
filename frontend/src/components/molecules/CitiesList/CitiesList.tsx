import { Component } from 'react';
import List from 'atoms/List';
import ListItem from 'atoms/ListItem';
import { Props } from './types';

export default class CitiesList extends Component<Props> {
  render() {
    return (
      <List>
        {['Ottawa', 'Moscow', 'Tokyo'].map((city) => (
          <ListItem key={city} active={city === this.props.active} onClick={this.props.setActive}>
            {city}
          </ListItem>
        ))}
      </List>
    );
  }
}
