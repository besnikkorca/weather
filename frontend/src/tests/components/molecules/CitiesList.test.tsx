import { render, screen } from '@testing-library/react';
import CitiesList from 'molecules/CitiesList';
import cities from '../../toydata/citiesList';

describe('CitiesList', () => {
  it('should render', () => {
    const view = render(<CitiesList cities={cities} active="Ottawa" setActive={jest.fn()} />);

    expect(view).toMatchSnapshot();
  });

  it('should render with active city', () => {
    const view = render(<CitiesList cities={cities} active="Ottawa" setActive={jest.fn()} />);

    expect(view).toMatchSnapshot();
  });

  it('should render with Moscow active', () => {
    const view = render(<CitiesList cities={cities} active="Moscow" setActive={jest.fn()} />);

    expect(view).toMatchSnapshot();
  });

  it('should render with Tokyo active', () => {
    const view = render(<CitiesList cities={cities} active="Tokyo" setActive={jest.fn()} />);

    expect(view).toMatchSnapshot();
  });

  it('should call setActive when clicked', () => {
    const mockSetActive = jest.fn();
    render(<CitiesList cities={cities} active="Ottawa" setActive={mockSetActive} />);

    screen.getByText('Moscow').click();

    expect(mockSetActive).toBeCalledTimes(1);
  });

  it('should call setActive with Moscow when clicked', () => {
    const mockSetActive = jest.fn();
    render(<CitiesList cities={cities} active="Ottawa" setActive={mockSetActive} />);

    screen.getByText('Moscow').click();

    expect(mockSetActive).toBeCalledWith('Moscow');
  });
});
