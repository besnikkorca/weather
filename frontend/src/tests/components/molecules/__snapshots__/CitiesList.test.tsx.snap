// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="list"
      >
        <li
          class="listItem active"
        >
          Ottawa
        </li>
        <li
          class="listItem notActive"
        >
          Moscow
        </li>
        <li
          class="listItem notActive"
        >
          Tokyo
        </li>
        <li
          class="listItem notActive"
        >
          London
        </li>
        <li
          class="listItem notActive"
        >
          New York
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="list"
    >
      <li
        class="listItem active"
      >
        Ottawa
      </li>
      <li
        class="listItem notActive"
      >
        Moscow
      </li>
      <li
        class="listItem notActive"
      >
        Tokyo
      </li>
      <li
        class="listItem notActive"
      >
        London
      </li>
      <li
        class="listItem notActive"
      >
        New York
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CitiesList should render with Moscow active 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="list"
      >
        <li
          class="listItem notActive"
        >
          Ottawa
        </li>
        <li
          class="listItem active"
        >
          Moscow
        </li>
        <li
          class="listItem notActive"
        >
          Tokyo
        </li>
        <li
          class="listItem notActive"
        >
          London
        </li>
        <li
          class="listItem notActive"
        >
          New York
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="list"
    >
      <li
        class="listItem notActive"
      >
        Ottawa
      </li>
      <li
        class="listItem active"
      >
        Moscow
      </li>
      <li
        class="listItem notActive"
      >
        Tokyo
      </li>
      <li
        class="listItem notActive"
      >
        London
      </li>
      <li
        class="listItem notActive"
      >
        New York
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CitiesList should render with Tokyo active 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="list"
      >
        <li
          class="listItem notActive"
        >
          Ottawa
        </li>
        <li
          class="listItem notActive"
        >
          Moscow
        </li>
        <li
          class="listItem active"
        >
          Tokyo
        </li>
        <li
          class="listItem notActive"
        >
          London
        </li>
        <li
          class="listItem notActive"
        >
          New York
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="list"
    >
      <li
        class="listItem notActive"
      >
        Ottawa
      </li>
      <li
        class="listItem notActive"
      >
        Moscow
      </li>
      <li
        class="listItem active"
      >
        Tokyo
      </li>
      <li
        class="listItem notActive"
      >
        London
      </li>
      <li
        class="listItem notActive"
      >
        New York
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CitiesList should render with active city 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="list"
      >
        <li
          class="listItem active"
        >
          Ottawa
        </li>
        <li
          class="listItem notActive"
        >
          Moscow
        </li>
        <li
          class="listItem notActive"
        >
          Tokyo
        </li>
        <li
          class="listItem notActive"
        >
          London
        </li>
        <li
          class="listItem notActive"
        >
          New York
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="list"
    >
      <li
        class="listItem active"
      >
        Ottawa
      </li>
      <li
        class="listItem notActive"
      >
        Moscow
      </li>
      <li
        class="listItem notActive"
      >
        Tokyo
      </li>
      <li
        class="listItem notActive"
      >
        London
      </li>
      <li
        class="listItem notActive"
      >
        New York
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
